homeassistant/components/danfoss_air/sensor.py

```diff
@@ -27,67 +27,89 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
             TEMP_CELSIUS,
             ReadCommand.exhaustTemperature,
             DEVICE_CLASS_TEMPERATURE,
+            STATE_CLASS_MEASUREMENT,
         ],
         [
             "Danfoss Air Outdoor Temperature",
             TEMP_CELSIUS,
             ReadCommand.outdoorTemperature,
             DEVICE_CLASS_TEMPERATURE,
+            STATE_CLASS_MEASUREMENT,
         ],
         [
             "Danfoss Air Supply Temperature",
             TEMP_CELSIUS,
             ReadCommand.supplyTemperature,
             DEVICE_CLASS_TEMPERATURE,
+            STATE_CLASS_MEASUREMENT,
         ],
         [
             "Danfoss Air Extract Temperature",
             TEMP_CELSIUS,
             ReadCommand.extractTemperature,
             DEVICE_CLASS_TEMPERATURE,
+            STATE_CLASS_MEASUREMENT,
         ],
         [
             "Danfoss Air Remaining Filter",
             PERCENTAGE,
             ReadCommand.filterPercent,
             None,
+            None,
         ],
         [
             "Danfoss Air Humidity",
             PERCENTAGE,
             ReadCommand.humidity,
             DEVICE_CLASS_HUMIDITY,
+            STATE_CLASS_MEASUREMENT,
+        ],
+        ["Danfoss Air Fan Step", PERCENTAGE, ReadCommand.fan_step, None, None],
+        [
+            "Danfoss Air Exhaust Fan Speed",
+            "RPM",
+            ReadCommand.exhaust_fan_speed,
+            None,
+            None,
+        ],
+        [
+            "Danfoss Air Supply Fan Speed",
+            "RPM",
+            ReadCommand.supply_fan_speed,
+            None,
+            None,
         ],
-        ["Danfoss Air Fan Step", PERCENTAGE, ReadCommand.fan_step, None],
-        ["Danfoss Air Exhaust Fan Speed", "RPM", ReadCommand.exhaust_fan_speed, None],
-        ["Danfoss Air Supply Fan Speed", "RPM", ReadCommand.supply_fan_speed, None],
         [
             "Danfoss Air Dial Battery",
             PERCENTAGE,
             ReadCommand.battery_percent,
             DEVICE_CLASS_BATTERY,
+            None,
         ],
     ]
 
     dev = []
 
     for sensor in sensors:
-        dev.append(DanfossAir(data, sensor[0], sensor[1], sensor[2], sensor[3]))
+        dev.append(
+            DanfossAir(data, sensor[0], sensor[1], sensor[2], sensor[3], sensor[4])
+        )
 
     add_entities(dev, True)
 
 
 class DanfossAir(SensorEntity):
     """Representation of a Sensor."""
 
-    def __init__(self, data, name, sensor_unit, sensor_type, device_class):
+    def __init__(self, data, name, sensor_unit, sensor_type, device_class, state_class):
         """Initialize the sensor."""
         self._data = data
         self._name = name
         self._state = None
         self._type = sensor_type
         self._unit = sensor_unit
         self._device_class = device_class
+        self._state_class = state_class
```

Comment:
```suggestion
        self._attr_state_class = state_class
```

https://developers.home-assistant.io/docs/core/entity#entity-class-or-instance-attributes